lizmap/modules/lizmap/classes/qgisForm.class.php

```diff
@@ -452,13 +452,85 @@ public function setFormDataFromFields($feature)
                 }
                 $form->setData($ref.'_hidden', $value);
             } else {
+                if (in_array(strtolower($this->formControls[$ref]->fieldEditType), array('date', 'time', 'datetime'))) {
+                    $edittype = $this->formControls[$ref]->getEditType();
+                    if ($edittype && property_exists($edittype, 'options')
+                            && property_exists($edittype->options, 'field_format')) {
+                        $format = $this->convertQgisFormatToPHP($edittype->options->field_format);
+                        $date = DateTime::createFromFormat($format, $value);
+                        $value = $date->format('Y-m-d H:i:s');
+                    }
+                }
                 $form->setData($ref, $value);
             }
         }
 
         return $form;
     }
 
+    public function getDateTimeConversionTab()
+    {
+        return array_reverse(array(
+            'd' => 'j',
+            'dd' => 'd',
+            'ddd' => 'D',
+            'dddd' => 'l',
+            'M' => 'n',
+            'MM' => 'm',
+            'MMM' => 'M',
+            'MMMM' => 'F',
+            'yy' => 'y',
+            'yyyy' => 'Y',
+            'H' => 'G',
+            'HH' => 'H',
+            'h' => 'G',
+            'hh' => 'H',
+            'AP' => 'A',
+            'ap' => 'a',
+            'm' => 'i',
+            'mm' => 'i',
+            'ss' => 's',
+            't' => 'T',
+        ));
+    }
+
+    public function convertQgisFormatToPHP($fieldFormat)
+    {
+        $dateFormat = $fieldFormat;
+        // convertion from QGIS to PHP format
+        $format = $this->getDateTimeConversionTab();
+        $format12h = array('a', 'ap', 'A', 'AP');
+        foreach ($format12h as $am) {
+            if (strstr($dateFormat, $am)) {
+                $format['h'] = 'g';
+                $format['hh'] = 'h';
+
+                break;
+            }
+        }
+        foreach ($format as $qgis => $php) {
+            $dateFormat = str_replace($qgis, $php, $dateFormat);
+        }
+
+        return $dateFormat;
+    }
+
+    /**
+     * Converts the datetime to the format specified in the qgis Project.
+     *
+     * @param string $value       The datetime to convert
+     * @param object $edittype    The format of the date
+     * @param mixed  $fieldFormat
```

Comment:
```suggestion
     * @param string $value           The datetime to convert
     * @param string $fieldFormat The format of the date
     * @return string
```
Update the docstring